### PR TITLE
Upgrade to micronaut 2.0

### PR DESCRIPTION
**Pull request by croudet**
and add netty metrics

### CODE REVIEW

File: gradle.properties [hunk 1]
```diff
@@ -4,9 +4,11 @@ micronautDocsVersion=1.0.23
 micrometerVersion=1.5.4
 validationVersion=2.0.1.Final
 jcacheVersion=1.1.1
-micronautVersion=1.3.5
+micronautVersion=2.0.0
 micronautTestVersion=1.2.2
 groovyVersion=2.5.11
+groovyVersion=3.0.4
+nettyVersion=4.1.50.Final
```
Comment (ilopmar) [inline gradle.properties:11]: Same as in #72. We should use `4.1.48.Final`

**Comment by ilopmar** [comment]
@croudet I've merged the PR for Micronaut 1.3.x and released version 2.1.0. Can you please fix the conflicts here and rebase master so we have this PR ready to merge? Thanks :)

**Comment by croudet** [comment]
I have rebased the branch.

**Comment by ilopmar** [comment]
Awesome! I'll merge this as soon as we have new branch for Micronaut 2.1 in core.

**Review by ilopmar — APPROVED** [review]

**Pull request merged**